apps/ledger-live-mobile/e2e/models/trade/swapFormPage.ts

```diff
@@ -16,44 +16,53 @@ export default class SwapFormPage {
   swapDestinationSelector = () => getElementById("swap-destination-selector");
   swapSourceInputTextbox = () => getElementById("swap-source-amount-textbox");
   exchangeButton = () => getElementById("exchange-button");
+  chooseProviderButton = () => getElementById("choose-provider-button");
   sendMaxToggle = () => getElementById("exchange-send-max-toggle");
   termsAcceptButton = () => getElementById("terms-accept-button");
   termsCloseButton = () => getElementById("terms-close-button");
 
-  async openViaDeeplink() {
-    await openDeeplink(baseLink);
+  openViaDeeplink() {
+    return openDeeplink(baseLink);
   }
 
-  async navigateToSwapForm() {
-    await tapByElement(this.swapFormTab());
+  navigateToSwapForm() {
+    return tapByElement(this.swapFormTab());
   }
 
-  async navigateToSwapHistory() {
-    await tapByElement(this.swapHistoryTab());
+  navigateToSwapHistory() {
+    return tapByElement(this.swapHistoryTab());
   }
 
-  async openSourceAccountSelector() {
-    await tapByElement(this.swapSourceSelector());
+  openSourceAccountSelector() {
+    return tapByElement(this.swapSourceSelector());
   }
 
-  async openDestinationAccountSelector() {
-    await tapByElement(this.swapDestinationSelector());
+  openDestinationAccountSelector() {
+    return tapByElement(this.swapDestinationSelector());
   }
 
-  async selectAccount(accountText: string) {
-    await tapByText(accountText);
+  selectAccount(accountText: string) {
+    return tapByText(accountText);
   }
 
   async enterSourceAmount(amount: string) {
     await clearTextByElement(this.swapSourceInputTextbox());
     await typeTextByElement(this.swapSourceInputTextbox(), amount);
   }
 
-  async sendMax() {
-    await tapByElement(this.sendMaxToggle());
+  goToProviderSelection() {
+    return tapByElement(this.chooseProviderButton());
   }
 
-  async startExchange() {
```

Comment:
We are already doing the `await` in the test file itself, so having an `await` in the models was redundant. Doing it this way means we are just passing the promise doing to the test and then `await`ing it there, instead of doing two awaits.